pypeit/scripts/identify.py

```diff
@@ -82,55 +92,236 @@ def main(args):
+
+            # Generate a map of the instrumental spectral FWHM
+            # TODO nsample should be a parameter
+            fwhm_map = autoid.map_fwhm(wavecal.msarc.image, wavecal.gpm, wavecal.slits_left, wavecal.slits_right, wavecal.slitmask,
+                                    nsample=10, slit_bpm=wavecal.wvc_bpm, specord=wavecal.par['fwhm_spec_order'],
+                                    spatord=wavecal.par['fwhm_spat_order'])
+            # Calculate the typical spectral FWHM down the centre of the slit
+            measured_fwhms = np.zeros(slits.nslits, dtype=object)
+            for islit in range(slits.nslits):
+                if islit not in ok_mask_idx:
+                    continue
+                # Measure the spectral FWHM (in pixels) at the midpoint of the slit
+                # (i.e. the midpoint in both the spectral and spatial directions)
+                measured_fwhms[islit] = fwhm_map[islit].eval(wavecal.msarc.image.shape[0]//2, 0.5)
+
+            # Save for redo's
+            wavecal.measured_fwhms = measured_fwhms
+            if args.new_sol:
+                sv_par = par.data.copy()
+                j_par = jsonify(sv_par)
+                strpar = json.dumps(j_par)
+
+                wv_calib = WaveCalib(wv_fits=None,
+                                    fwhm_map=fwhm_map,
+                                    arc_spectra=np.array(wavecal.extract_arcs()[0]),
+                                    nslits=wavecal.slits.nslits,
+                                    spat_ids=wavecal.slits.spat_id,
+                                    PYP_SPEC=wavecal.spectrograph.name,
+                                    lamps=','.join(wavecal.lamps),
+                                    strpar = strpar)
+
+
+
+            #iterate over each slit to add to the wv_calib
+            if args.new_sol:
+                slits_inds = np.arange(slits.nslits)
+            else:
+                if args.slits == 'all':
+                    slits_inds = np.arange(slits.nslits)
+                else:
+                    slit_list = list(args.slits)
+                    ii = 0
+                    slits_inds = []
+                    while 2*ii+1 < len(slit_list):
+                        slits_inds.append(int(slit_list[2*ii+1]))
+                        ii += 1
+                    slits_inds = np.array(slits_inds)
+            fits_dicts = []
+            specdata = []
+            wv_fits_arr = []
+            lines_pix_arr = []
+            lines_wav_arr = []
+            lines_fit_ord = []
+            custom_wav = []
+            custom_wav_ind = []
+            #print(args.slits, slits_inds)
+            for slit_val in slits_inds:
+                # Load the calibration frame (if it exists and is desired).  Bad-pixel mask
+                # set to any flagged pixel in Arc.
+                wv_calib_slit = None
+                if wv_calib:
```

Comment:
Style nit-pick: I'd prefer that we be explicit by using `if wv_calib is not None`.